projects/js-packages/publicize-components/src/components/form/index.tsx

```diff
@@ -25,8 +29,22 @@ import { SharePostForm } from './share-post-form';
  * @return {object} - Publicize form component.
  */
 export default function PublicizeForm() {
-	const { hasConnections, hasEnabledConnections } = useSocialMediaConnections();
+	const { hasConnections, hasEnabledConnections, connections } = useSocialMediaConnections();
 	const { isPublicizeEnabled, isPublicizeDisabledBySitePlan } = usePublicizeConfig();
+	const { attachedMedia } = useAttachedMedia();
+	const featuredImageId = useFeaturedImage();
+
+	const mediaId = attachedMedia[ 0 ]?.id || featuredImageId;
+	const { validationErrors, isConvertible } = useMediaRestrictions(
+		connections,
+		useMediaDetails( mediaId )[ 0 ]
+	);
+
+	const showSharePostForm =
+		isPublicizeEnabled &&
+		( hasEnabledConnections ||
+			attachedMedia.length > 0 ||
+			( Object.keys( validationErrors ).length !== 0 && ! isConvertible ) );
```

Comment:
In what case will there no validation errors when the media is not convertible?

I mean if the media is not convertible, won't there already be validation errors?

Reply:
Good point.

Reply:
So you say we should only look for `! isConvertible` in that part of the logic?

Reply:
I am not sure but my concern is that if the media is not convertible, we should already show/have validation errors. If that is not the case then we may need to fix that first, otherwise, yes, we may only look for `! isConvertible` in that part of the logic.

Reply:
> I am not sure but my concern is that if the media is not convertible, we should already show/have validation errors.

Yeah we do, I think it's true that if something is not convertible, that means we have at least 1 error

Reply:
Cool. Thank you for the confirmation.